tests/conftest.py

```diff
@@ -530,6 +530,7 @@ def s3():
     return s3fs.S3FileSystem(
         key=os.environ.get("AWS_ACCESS_KEY_ID"),
         secret=os.environ.get("AWS_SECRET_ACCESS_KEY"),
+        requester_pays=True,
```

Comment:
How does this affect the tests that do not require this?

Reply:
My guess is we'll pay for more transfer fees (https://s3fs.readthedocs.io/en/latest/#requester-pays-buckets) -- though this is probably already happening for datasets in the Coiled AWS account. Some public datasets require us to specify `requester_pays=True`. We didn't have a mechanism for passing `s3fs` options through, so I just hard coded it. cc @ntabris for thoughts on if this is problematic

Reply:
not sure if it's problematic. this feels like something that would make sense to have configurable per-test though, right? (that feels to me like the easier way of making sure it isn't more problematic, but of course, it would be someone else doing that work so of course it feels easier to me.)

Reply:
I agree with Nat, the only test that needs `requester_pays` is the one you are adding here.

Reply:
I've added a new `s3_factory` fixture for creating non-default `S3FileSystem` instances. Note it's function scoped instead of session scoped like the existing `s3` fixture (changing `s3` to be function scoped breaks other fixtures where `s3` is currently used)